### PR TITLE
Bug fix: disabling all items in a menu results in failed build

### PR DESCRIPTION
**Pull request by tqureshi-uog**
# Summary of changes
Add a proper null check to the header menu component along with other upgrades and improvements

## Frontend
- Update menu graphql query in `page.js` to retrieve enabled items only
- Remove deprecated `StaticQuery` function in `headerMenu.js` and replace with `useStaticQuery`
- Add null check for page title in `headerMenu.js`

### Incidental changes
- Upgrade packages to latest versions (so other plugins are in line with already-updated `gatsby-source-wordpress`)
- Remove unneeded `contentExists` function from `hero.js`

[ x ] My changes are accessible (at minimum WCAG 2.0 Level AA)
[ x ] My changes are responsive and appear as expected on mobile and desktop views.
[ x ] After merging, I will update the [Content Hub documentation](https://uoguelphca.sharepoint.com/sites/UniversityContentHubInformationGroup) and ensure the Content Hub trainer understands how my changes will affect users.

## Backend
N/A

# Test Plan
This PR is set to use https://api.testugconthub.uoguelph.dev/ as the backend where, for test purposes, all items in the `Chancellor` menu were disabled at the time this feature branch was built.

1. Go to https://dismenu--test-ugconthub.netlify.app/chancellor and verify the page loads successfully (the usual menu will not appear)
2. Do the same for https://dismenu--test-ugconthub.netlify.app/chancellor/contact/
3. Go to https://dismenu--test-ugconthub.netlify.app/grce and verify its menu (which was left untouched) still appears as expected
4. Do the same for https://dismenu--test-ugconthub.netlify.app/quality-assurance

### CODE REVIEW

**Review by ccswdscoop — APPROVED** [review]
Looks good on my end.

All pages loaded as per your test cases.

File: package.json
Comment (mmafe) [inline package.json:17]: In future, I'd recommend doing the package update on a separate PR. If something goes wrong, it becomes more difficult to determine the cause if we're packaging all the changes together.

File: src/components/shared/headerMenu.js
Comment (mmafe) [inline src/components/shared/headerMenu.js:87]: Should we use a strict inequality check here? `!==`

**Review by mmafe — CHANGES_REQUESTED** [review]
**Overall:** Fix works as expected 👍 
**Main requested change:** Wondering about whether we should use a strict inequality null check on headerMenu.js.

File: package.json
Comment (mmafe) [reply package.json:17]: I'm wondering if the package updates were the reason the Netlify build was taking so long to complete.

We still have to create a release branch potentially from both #203 and this one, so I'm wondering if we:
- Put the menu change onto a new PR
- Put the package updates + hero.js change onto a new PR
- That way you can merge them one at a time onto develop and make sure they run on Netlify before creating a release branch from develop on Monday.

File: src/components/shared/headerMenu.js
Comment (tqureshi-uog) [reply src/components/shared/headerMenu.js:87]: I chose not to because I wanted the statement to check for both `null` _and_ `undefined` - if I'd used the strict inequality check, the statement would work only for `null` - at least, that's what Copilot told me :)

> In JavaScript, != and !== are two different types of comparison operators that have a subtle but important difference:
> 
> != is the non-strict equality operator. It compares the values of the two operands for equality after performing any necessary type conversions. **This means that if you compare a value with null using !=, it will also return true for undefined values because null and undefined are loosely equal.** So, pageHome.title != null will be true if pageHome.title is either null or undefined.
>
> !== is the strict non-equality operator. It compares both the value and the type of the two operands. No type conversion is done, and if the values are of different types, the values are considered unequal. So, pageHome.title !== null will only be true if pageHome.title is not exactly null. If pageHome.title is undefined, the comparison pageHome.title !== null will return false because null and undefined are of different types.

Comment (mmafe) [reply src/components/shared/headerMenu.js:87]: Ahhhh, this makes sense.

File: package.json
Comment (mmafe) [reply package.json:17]: I'm creating a develop Netlify environment, merging the first PR (#203) onto develop (see if it completes), then will merge #204 onto develop and see if it completes.

**Comment by anduonguog** [comment]
The pages passed the test case and were also tested for accessibility and responsiveness. They look good.

File: package.json
Comment (mmafe) [reply package.json:17]: @tqureshi-uog The develop branch did not finish within the time limit after merging #204. Would you be able to separate the PR into two PRs?

- One with the menu change
- One with the package updates + hero.js change

Then we can reset develop, merge #203, then merge the menu change and see if it completes without issue

Comment (mmafe) [reply package.json:17]: Actually...hang on. It looks like it almost finished. I'm going to re-run develop one more time (it's building the preview site)...maybe the cache will help it finish the next time it runs.

Comment (mmafe) [reply package.json:17]: @tqureshi-uog - Develop (with #203 and #204 merged) deployed successfully the second time around. 👍

**Review by mmafe — APPROVED** [review]

**Pull request merged**
